packages/form-js-viewer/src/Form.js

```diff
@@ -168,9 +168,12 @@ export class Form {
 
     const errors = this.validate();
 
+    const files = this.get('fileRegistry').getAllFiles();
```

Comment:
I think this needs to be filtered based on the available instances.  Otherwise it'll submit hidden files as well. 

Lazy pseudocode:

```
const filteredInstances = instances
  .filter(instance => instance.type === 'filepicker')
  .map(instance => instance.value);

const filteredFiles = files.filter(file => filteredInstances.includes(file.value));
```

But **first** there's the issue of how these files are keyed (see other comment).

Reply:
I deleted the files when the component unmounts, this the logic is encapsulated on the component